.circleci/config.yml

```diff
@@ -0,0 +1,56 @@
+version: 2
+jobs:
+  test:
+    docker:
+      - image: circleci/node:4-browsers
+    steps:
+      - checkout
```

Comment:
I don't know if that's a big improvement, but we can have a "setup" job with `checkout`:
```yaml
jobs:
  setup:
     docker: # image node:4...
     steps:
        - checkout
        - make node_modules (or the equivalent for this repo)
        - persist_to_workspace:
          root: .
          paths: [.]
   test: # image .....
      steps:
         - attach_workspace: { at: . }
         - run: ...
   publish: # image ...
       steps:
         - attach_workspace: { at: . }
         - run: ...

workflows:
  jobs:
    - setup
    - test:
      requires: [setup]
    - publish:
       requires: [setup, publish]
```

Reply:
It would still have to run it twice

Reply:
Oh I see what you mean. Yeah I think because the setup is so small, duplicating code is probably ok.